ob1k-cache/src/main/java/com/outbrain/ob1k/cache/LocalAsyncCache.java

```diff
@@ -112,9 +112,22 @@ public LocalAsyncCache() {
 
   private ComposableFuture<V> loadElement(final K key) {
     final ComposableFuture<V> loadedElement = loader.load(cacheName, key).materialize();
-    return loadedElement.continueOnError((FutureErrorHandler<V>) error -> {
-      loadingCache.asMap().remove(key, loadedElement);
-      return fromError(error);
+    return loadedElement;
+  }
+
+  private ComposableFuture<V> extractMapEntry(ComposableFuture<Map<K, V>> loaded ,K key) {
+    return loaded
+      .continueOnSuccess((FutureSuccessHandler<Map<K, V>, V>) loaderResults -> {
+      final V res = loaderResults.get(key);
+      if (res != null) {
+        return fromValue(res);
+      } else {
+        if (failOnMissingEntries) {
```

Comment:
failOnMissingEntries is always true, and cannot be changed outside. Maybe just removing this redundant flag?